allensdk/brain_observatory/behavior/eye_tracking_processing.py

```diff
@@ -207,10 +207,16 @@ def process_eye_tracking_data(eye_data: pd.DataFrame,
                                             outliers,
                                             dilation_frames=dilation_frames)
 
+    # remove outliers/likely blinks from default `pupil_area` column
+    # maintain full fits in the `pupil_area_raw` column
+    pupil_area_raw = pupil_areas.copy()
+    pupil_areas[likely_blinks == True] = np.nan
+
     eye_data.insert(0, "time", frame_times)
     eye_data.insert(1, "cr_area", cr_areas)
     eye_data.insert(2, "eye_area", eye_areas)
     eye_data.insert(3, "pupil_area", pupil_areas)
     eye_data.insert(4, "likely_blink", likely_blinks)
+    eye_data.insert(5, "pupil_area_raw", pupil_area_raw)
```

Comment:
I believe this raw area needs to be added to the NWB file:

https://github.com/AllenInstitute/AllenSDK/blob/17d8175bda73d9b367372df79d049ccb79449023/allensdk/brain_observatory/behavior/session_apis/data_io/behavior_ophys_nwb_api.py#L485

Reply:
@matchings the nwb file has three places that share the same `EllipseSeries` data structure:
`eye_tracking`, `pupil_tracking`, and `corneal_reflection_tracking`.
You're asking here for `pupil_tracking` to have a new `area_raw` field.
I can:
* define a new nwb thing called `PupilEllipseSeries` that has the `area_raw`, leaving the other 2 unchanged,
* make `eye_tracking` and `corneal_reflection_tracking` have `area_raw` but be NaN or same values as `area`.
* propagate Doug's changes in the SDK for blink masking to these other two `EllipseSeries` so that all 3 have `area` and `area_raw`

Reply:
or @njmei can give me an nwb hint about making an nwb field optional.

Reply:
discussed offline. applying to all 3 fields.

Reply:
this is me confirming that im good with applying blink correction to all _area fields, as long as there is also a _raw version of it that is not blink corrected. @dougollerenshaw any objections to that?

Reply:
do the timeseries that will be in the _raw columns undergo any other filtering / processing other than the blink correction? just wondering how literal "raw" means here

Reply:
It is working from this 'eye_data':

    Parameters
    ----------
    eye_data : pd.DataFrame
        A 'raw' eye tracking dataframe produced by load_eye_tracking_hdf()
        
[link](https://github.com/AllenInstitute/AllenSDK/blob/17d8175bda73d9b367372df79d049ccb79449023/allensdk/brain_observatory/behavior/eye_tracking_processing.py#L160-L163)

Reply:
thanks @djkapner that was helpful. looks like no additional processing to me, so it should truly be raw data